enhancements/security/internal-pki-config.md

```diff
@@ -0,0 +1,1695 @@
+}
+
+// PKISpec uses a parameterized union pattern where the type field determines
+// the validity and interpretation of the defaults, categories, and overrides fields.
+//
+// +kubebuilder:validation:XValidation:rule="self.type == 'Unmanaged' ? (!has(self.defaults) && !has(self.categories) && !has(self.overrides)) : true",message="defaults, categories, and overrides must not be set when type is Unmanaged"
+// +kubebuilder:validation:XValidation:rule="self.type == 'Default' ? (!has(self.defaults) && !has(self.categories) && !has(self.overrides)) : true",message="defaults, categories, and overrides must not be set when type is Default"
+// +union
+type PKISpec struct {
+    // type determines how PKI configuration is managed.
+    //
+    // - Unmanaged: Components use their existing hardcoded certificate generation behavior, exactly as if this feature did not exist.
+    //   Each component generates certificates using whatever parameters it was using before this feature.
+    //   While most components use RSA 2048, some may use different parameters.
+    //   Use of this mode might prevent upgrading to the next major OpenShift release.
+    // + Default when upgrading from a version without this feature to ensure zero behavior change.
+    //   The defaults, categories, and overrides fields must not be set.
+    //
+    // - Default: Use OpenShift-recommended best practices for certificate generation.
+    //   The specific parameters may evolve across OpenShift releases to adopt improved cryptographic standards.
+    //   In the initial release, this matches Unmanaged behavior for each component.
+    //   In future releases, this may adopt ECDSA or larger RSA keys based on industry best practices.
+    //   Recommended for most customers who want to benefit from security improvements automatically.
+    // + Default when installing a fresh cluster.
+    //   The defaults, categories, and overrides fields must not be set.
+    //
+    // - Custom: Administrator explicitly configures cryptographic parameters.
+    // + Recommended for customers with specific compliance requirements or organizational PKI policies.
+    // 
+    // + When upgrading from a version without this feature:
+    // + - The PKI resource is created with type: Unmanaged to ensure zero behavior change.
+    // +
+    // + When installing a fresh cluster:
+    // + - If no PKI configuration is provided in install-config.yaml, type: Default is used.
+    // + - If PKI configuration is provided in install-config.yaml, type: Custom is used with the specified configuration.
+    //
+    // +kubebuilder:validation:Required
+    // +kubebuilder:validation:Enum=Unmanaged;Default;Custom
+    // +unionDiscriminator
+    Type PKIManagementType `json:"type"`
+
+    // Embed PKIProfile to provide reusable certificate parameter fields.
+    // Go API utilities can use PKIProfile to programmatically define custom profiles.
+    PKIProfile `json:",inline"`
```

Comment:
For discriminated unions we generally encourage the use of a field whose name matches the discriminator. This allows us to expand in the future to support other values that may require configuration without causing an overlap for the fields that need to be supplied.

I'd expect this to end up looking something like:
```yaml
type: Custom
custom:
  ...
```